packages/autocomplete-shared/src/invariant.ts

```diff
@@ -3,12 +3,17 @@
  * This is used to make development a better experience to provide guidance as
  * to where the error comes from.
  */
-export function invariant(condition: boolean, message: string) {
+export function invariant(
+  condition: boolean,
+  message: string | (() => string)
+) {
   if (!__DEV__) {
     return;
   }
 
   if (!condition) {
-    throw new Error(`[Autocomplete] ${message}`);
+    const _message = typeof message === 'string' ? message : message();
+
+    throw new Error(`[Autocomplete] ${_message}`);
```

Comment:
Function is the particular case so I would reverse the assertion to only execute it when it's actually a function (and not when it's something else than a string).

To avoid the `_` prefix, I think it's fine to inline it too. Wdyt?

```suggestion
    throw new Error(`[Autocomplete] ${typeof message === 'function' ? message() : message}`);
```

Reply:
Makes sense!